ios/RNTrackPlayer/RNTrackPlayer.swift

```diff
@@ -263,24 +263,24 @@ public class RNTrackPlayer: RCTEventEmitter {
     }
     
     @objc(updateOptions:resolver:rejecter:)
-    public func update(options: [String: Any], resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
-
-        var capabilitiesStr = options["capabilities"] as? [String] ?? []
-        if (capabilitiesStr.contains("play") && capabilitiesStr.contains("pause")) {
-            capabilitiesStr.append("togglePlayPause");
-        }
-        let capabilities = capabilitiesStr.compactMap { Capability(rawValue: $0) }
-        
-        let remoteCommands = capabilities.map { capability in
-            capability.mapToPlayerCommand(jumpInterval: options["jumpInterval"] as? NSNumber,
-                                          likeOptions: options["likeOptions"] as? [String: Any],
-                                          dislikeOptions: options["dislikeOptions"] as? [String: Any],
-                                          bookmarkOptions: options["bookmarkOptions"] as? [String: Any])
+    public func update(options: [String: Any], resolve: @escaping RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
+        DispatchQueue.main.async {
+            var capabilitiesStr = options["capabilities"] as? [String] ?? []
+            if (capabilitiesStr.contains("play") && capabilitiesStr.contains("pause")) {
+                capabilitiesStr.append("togglePlayPause");
+            }
+            let capabilities = capabilitiesStr.compactMap { Capability(rawValue: $0) }
+            
+            let remoteCommands = capabilities.map { capability in
+                capability.mapToPlayerCommand(jumpInterval: options["jumpInterval"] as? NSNumber,
+                                              likeOptions: options["likeOptions"] as? [String: Any],
+                                              dislikeOptions: options["dislikeOptions"] as? [String: Any],
+                                              bookmarkOptions: options["bookmarkOptions"] as? [String: Any])
+            }
+            self.player.enableRemoteCommands(remoteCommands)
+            
```

Comment:
```swift
self.player.remoteCommands = remoteCommands
```
The remoteCommands variable also needs to be set, otherwise, it may be reset by self.enableRemoteCommands(remoteCommands) here  https://github.com/react-native-kit/react-native-track-player/blob/2392845c0ea3663da891835a48a9ea3ded3fa01b/ios/RNTrackPlayer/Vendor/SwiftAudio/Classes/AudioPlayer.swift#L226

Reply:
That functionality is disabled in this overridden function: https://github.com/react-native-kit/react-native-track-player/blob/2392845c0ea3663da891835a48a9ea3ded3fa01b/ios/RNTrackPlayer/RNTrackPlayerAudioPlayer.swift#L119

Reply:
> That functionality is disabled in this overridden function:
> 
> https://github.com/react-native-kit/react-native-track-player/blob/2392845c0ea3663da891835a48a9ea3ded3fa01b/ios/RNTrackPlayer/RNTrackPlayerAudioPlayer.swift#L119

You are right, thank you for pointing this out